HexCtrl/src/CHexCtrl.cpp

```diff
@@ -5195,4 +5202,23 @@ void CHexCtrl::OnVScroll(UINT /*nSBCode*/, UINT /*nPos*/, CScrollBar* /*pScrollB
 		RedrawWindow();
 		ParentNotify(HEXCTRL_MSG_VIEWCHANGE); //Indicates to parent that view has changed.
 	}
+}
+
+DWORD CHexCtrl::GetCurrentDateFormat()const
+{
+	return m_dwDateFormat;
+}
+
+void CHexCtrl::SetCurrentDateFormat(DWORD dwDateFormat)
+{
+	switch (dwDateFormat)
+	{
+	case 0:	//0=Month-Day-Year
+	case 2:	//2=Year-Month-Day
+		m_dwDateFormat = dwDateFormat;
+		break;
+	default: //1=Day-Month-Year (default)
+		m_dwDateFormat = 1;
+		break;
+	}

```

Comment:
I think it's not generally correct. In my view the `SetDateFormat` method must accept four different values:
`0, 1, 2` and, let's say, `-1` as a default. In all other cases it's an error/assert.
When `-1` is provided we fall back to the **current user default date format**.
```cpp
void CHexCtrl::SetDateFormat(DWORD dwDateFormat)
{
 if(dwDateFormat != -1 || dwDateFormat > 2)
    return;

 if(dwDateFormat==-1)
   {
       //m_dwDateFormat = GetUserDefault(...)
   }
...
}

Reply:
Added support for default using -1